coap/api/transport.go

```diff
@@ -166,10 +169,10 @@ func decodeMessage(msg *mux.Message) (*messaging.Message, error) {
 }
 
 func parseKey(msg *mux.Message) (string, error) {
-	if obs, _ := msg.Options.Observe(); obs != 0 && msg.Code == codes.GET {
+	if obs, _ := msg.Observe(); obs != 0 && msg.Code() == codes.GET {
 		return "", nil
```

Comment:
@1998-felix  I have tested with new coap-cli, it throws error while canceling the observer(unbscribing). 
So better we can remove this if statement. I hope this statement is no more needed.

```json
{"time":"2024-05-04T11:33:01.206749191Z","level":"WARN","msg":"Unsubscribe failed to complete successfully","duration":"1.37521ms","channel_id":"795b0be8-f8d9-4101-a40f-859ed573c71c","error":{"error":"failed to perform authorization over the entity","message":"failed to perform authorization over the entity"}}
```